include/grpc/support/slice.h

```diff
@@ -175,6 +175,10 @@ gpr_slice gpr_empty_slice(void);
 int gpr_slice_cmp(gpr_slice a, gpr_slice b);
 int gpr_slice_str_cmp(gpr_slice a, const char *b);
 
+/* Returns a c-string representation of a slice. Useful for logging or debugging
+   purposes only. The string representation can change in future */
+char *gpr_slice_to_cstring(gpr_slice slice);
```

Comment:
Do we need this? I am not sure whether we can do something like `%.*s` for gpr_log to log a slice.